config.nix

```diff
@@ -19,44 +19,52 @@
+         ++ [
+          ./patches/ghc/ghc-add-keepCAFs-to-rts.patch                         # https://gitlab.haskell.org/ghc/ghc/merge_requests/950  -- open
+          ./patches/ghc/lowercase-8.6.patch                                   # https://gitlab.haskell.org/ghc/ghc/merge_requests/949  -- merged; ghc-8.8.1
+          ./patches/ghc/dll-loader-8.4.2.patch                                # https://gitlab.haskell.org/ghc/ghc/merge_requests/949  -- open
+          ./patches/ghc/0001-Stop-the-linker-panic.patch                      # https://phabricator.haskell.org/D5012                  -- merged; ghc-8.8.1
+          ./patches/ghc/ghc-8.4.3-Cabal2201-no-hackage-tests.patch            # ?
+          ./patches/ghc/ghc-8.4.3-Cabal2201-allow-test-wrapper.patch          # https://github.com/haskell/cabal/pulls/5995            -- merged; cabal-3.0.0 (ghc-8.8.1)
+          ./patches/ghc/ghc-8.4.3-Cabal2201-response-file-support.patch       # https://github.com/haskell/cabal/pulls/5996            -- merged; cabal-3.0.0 (ghc-8.8.1)
+          ./patches/ghc/ghc-8.6-Cabal-fix-datadir.patch                       # https://github.com/haskell/cabal/issues/5862
+          ./patches/ghc/MR196--ghc-pkg-shut-up.patch                          # https://gitlab.haskell.org/ghc/ghc/merge_requests/196  -- merged; ghc-8.8.1
+          ./patches/ghc/MR948--32bit-cross-th.patch                           # https://gitlab.haskell.org/ghc/ghc/merge_requests/948  -- open
+         ]
+
+         # Patches for specific ghc versions.
+         ++ lib.optional (builtins.compareVersions drv.version "8.6.3" == 0)  ./patches/ghc/T16057--ghci-doa-on-windows.patch
+         ++ lib.optional (builtins.compareVersions drv.version "8.6.3" == 0)  ./patches/ghc/ghc-8.6.3-reinstallable-lib-ghc.patch
+         ++ lib.optional (builtins.compareVersions drv.version "8.6.4" == 0)  ./patches/ghc/ghc-8.6.4-reinstallable-lib-ghc.patch
          ++ lib.optional (builtins.compareVersions drv.version "8.6.4" == 0)  ./patches/ghc/ghc-8.6.4-reenable-th-qq-in-stage1.patch
          ++ lib.optional (builtins.compareVersions drv.version "8.6.4" == 0)  ./patches/ghc/ghc-8.6.4-better-plusSimplCountErrors.patch
-         ++ [
-          ./patches/ghc/ghc-add-keepCAFs-to-rts.patch
-          ./patches/ghc/lowercase-8.6.patch
-          ./patches/ghc/dll-loader-8.4.2.patch
-          ./patches/ghc/0001-Stop-the-linker-panic.patch
-          ./patches/ghc/ghc-8.4.3-Cabal2201-no-hackage-tests.patch
-          ./patches/ghc/ghc-8.4.3-Cabal2201-allow-test-wrapper.patch
-          ./patches/ghc/ghc-8.4.3-Cabal2201-response-file-support.patch
-          ./patches/ghc/ghc-8.6-Cabal-fix-datadir.patch
-          ./patches/ghc/MR196--ghc-pkg-shut-up.patch
-         ];
-        postPatch = (drv.postPath or "") + ''
-        autoreconf
-        '';
+         ;
+        postPatch = (drv.postPatch or "") + "\n" + "autoreconf";
```

Comment:
Can we get a comment about why we need to autoreconf? Shouldn't the configure phase happen after the patch phase?

Reply:
Right, but autoreconf, will regenerate the configure script if the .ac files were changed.

Reply:
So the normal GHC patch phase doesn't autoreconf? I guess if it doesn't do that we have to do it ourselves.

Reply:
er, configure phase

Reply:
Or we could put `autoreconfHook` in `nativeBuildInputs`, perhaps.

Reply:
It looks like it's already there :thinking:

Reply:
The underlying reason, and I should and will add a comment is that we may patch the .ac files.  So it's more a fallout from our gracious patching rather than something standard.

Reply:
TIL about `autoreconfHook`, I don't see it in the `nativeBuildInputs` of ghc expressions though.

Reply:
Intriguing: in nixpkgs it appears in the [8.2.2 derivation](https://github.com/input-output-hk/nixpkgs/blob/iohk-18.09/pkgs/development/compilers/ghc/8.2.2.nix#L216) but not the later ones. I have no idea what's gong on here. We could maybe use it but the existing thing is fine.

Reply:
Maybe they used to apply a patch that required the autoreconfHook, and when that patch wasn’t required anymore they dropped it.